File: Python/0205_isomorphic_strings/solution.py
def isomorphic_string(s,t):
    refer_dict = dict()
    for index in range(len(s)):
        if s[index] not in refer_dict.keys() and t[index] not in refer_dict.values():
            refer_dict[s[index]] = t[index]
    result = str()
    for letter in s:
        if letter in refer_dict:
            result += refer_dict[letter]
    if result == t:
        return True
    return False

s = "egg" 
t = "add"
print(isomorphic_string(s,t))

File: JavaScript/0205_isomorphic_strings/solution.js
function isomorphic_string(s,t) {
    let refer_dict = new Object();
    for (let index = 0; index < s.length; index ++) {
        if (!Object.keys(refer_dict).includes(s[index]) && !Object.values(refer_dict).includes(t[index])) {
            refer_dict[s[index]] = t[index];
        }
    }
    result = new String();
    for (let letter of s) {
        if (letter in refer_dict) {
            result = result + refer_dict[letter];
        }
    }
    if (result === t) {
        return true;
    }
    return false;
}

const s = "egg" 
const t = "add"
console.log(isomorphic_string(s,t))